File: examples/weather/src/test/java/com/example/weather/TestWeatherProvider.java
/*
 * Copyright (c) 2026 Konstantin Pavlov and contributors.
 */

package com.example.weather;

import com.example.weather.spi.CityNotFoundException;
import com.example.weather.spi.WeatherObservation;
import com.example.weather.spi.WeatherProvider;

import java.util.concurrent.CompletableFuture;

final class TestWeatherProvider implements WeatherProvider {

    @Override
    public WeatherObservation currentWeather(String city) throws CityNotFoundException {
        if ("Unknown".equals(city)) {
            throw new CityNotFoundException(city);
        }
        return new WeatherObservation(city, "Clear sky", 18.5, 52, 12.0);
    }

    @Override
    public CompletableFuture<WeatherObservation> currentWeatherAsync(String city) {
        try {
            return CompletableFuture.completedFuture(currentWeather(city));
        } catch (CityNotFoundException e) {
            return CompletableFuture.failedFuture(e);
        }
    }
}

File: examples/weather-mcp-kotlin/src/test/kotlin/com/example/weather/TestWeatherProvider.kt
/*
 * Copyright (c) 2026 Konstantin Pavlov/IT Staff and contributors.
 */

package com.example.weather

import com.example.weather.model.TemperatureUnit
import com.example.weather.spi.CityNotFoundException
import com.example.weather.spi.WeatherObservation
import com.example.weather.spi.WeatherProvider

class TestWeatherProvider : WeatherProvider {
    override fun currentWeather(
        city: String,
        temperatureUnit: TemperatureUnit,
    ): WeatherObservation {
        if (city == "Unknown") throw CityNotFoundException(city)
        return WeatherObservation(
            condition = "Clear sky",
            temperature = 18.5,
            temperatureUnit = temperatureUnit,
            humidity = 52,
            windSpeed = 12.0,
        )
    }
}
